project/resources/views/auth/roles.blade.php

```diff
@@ -10,7 +10,8 @@
     <div class="py-12">
         <div class="mx-auto sm:px-6 lg:px-8">
             <div class="bg-white overflow-hidden shadow-xl sm:rounded-lg">
-                <form method="POST">
+                <form method="POST" action="{{url("permissions")}}">
+                    @csrf
```

Comment:
Dobrze że dodałeś tutaj csrf, ale wtedy chyba te kolejne przed foreach nie są potrzebne no nie? Wydaje mi się, że to chroni cały form

Reply:
Tak, raczej tak, gdzieś mi się tam musiało pokopiować to nadmiarowo